pkg/addon/template.go

```diff
@@ -216,6 +218,18 @@ type MLASettings struct {
 	LoggingEnabled bool
 }
 
+type CSIOptions struct {
+
+	//vsphere
+	// StoragePolicy is the storage policy to use for vsphere csi addon
+	StoragePolicy string
+
+	//nutanix
+	StorageContainer        string
+	Fstype                  string
+	SsSegmentedIscsiNetwork *bool
+}
```

Comment:
Can we maybe change this to something like:

```go
type CSIOptions struct {
     Vsphere *VsphereCSIOptions
     Nutanix *NutanixCSIOptions
}

type VsphereCSIOptions struct {
    StoragePolicy string
}

type NutanixCSIOptions struct {
    StorageContainer        string
    Fstype                  string
    SsSegmentedIscsiNetwork *bool
}
```

I don't think it's a good idea to have a shared struct holding all provider-specific options.

Reply:
As this is only used for the templating and for now we do not have a lot of these features its okay in my opinion.

Reply:
Not a huge fan of it to be honest, but okay :D